src/_P039_Thermocouple.ino

```diff
@@ -30,28 +30,148 @@
 // Thermocouple Infos
 // http://www.bristolwatch.com/ele2/therc.htm
 
+// Resistor Temperature Detector Infos
+// https://en.wikipedia.org/wiki/Resistance_thermometer
+
 // Chips
 // MAX6675  - Cold-Junction-Compensated K-Thermocouple-to-Digital Converter (   0°C to +1024°C)
 //            https://cdn-shop.adafruit.com/datasheets/MAX6675.pdf (only
 // MAX31855 - Cold-Junction Compensated Thermocouple-to-Digital Converter   (-270°C to +1800°C)
 //            https://cdn-shop.adafruit.com/datasheets/MAX31855.pdf
 // MAX31856 - Precision Thermocouple to Digital Converter with Linearization   (-210°C to +1800°C)
 //            https://datasheets.maximintegrated.com/en/ds/MAX31856.pdf
+// MAX31865 - Precision Resistor Temperature Detector to Digital Converter with Linearization   (PT100 / PT1000)
+//            https://datasheets.maximintegrated.com/en/ds/MAX31865.pdf
+// TI Digital Temperature sensors with SPI interface
+//            https://www.ti.com/sensors/temperature-sensors/digital/products.html#p1918=SPI,%20Microwire
+// TI LM7x -  Digital temperature sensor with SPI interface 
+//            https://www.ti.com/lit/gpn/LM70
+//            https://www.ti.com/lit/gpn/LM71
+//            https://www.ti.com/lit/gpn/LM70
+//            https://www.ti.com/lit/gpn/LM74
+// TI TMP12x  Digital temperature sensor with SPI interface 
+//            https://www.ti.com/lit/gpn/TMP121
+//            https://www.ti.com/lit/gpn/TMP122
+//            https://www.ti.com/lit/gpn/TMP123
+//            https://www.ti.com/lit/gpn/TMP124
 
 # include <SPI.h>
 
+// // plugin-local quick activation of debug messages
+// #ifdef BUILD_NO_DEBUG
+//   #undef BUILD_NO_DEBUG
+// #endif
 
 # define PLUGIN_039
 # define PLUGIN_ID_039         39
-# define PLUGIN_NAME_039       "Environment - Thermocouple"
+# define PLUGIN_NAME_039       "Environment - Thermosensors"
 # define PLUGIN_VALUENAME1_039 "Temperature"
 
-# define P039_MAX_TYPE         PCONFIG(0)
-# define P039_TC_TYPE          PCONFIG(1)
-
-# define P039_MAX_6675         1
-# define P039_MAX_31855        2
-# define P039_MAX_31856        3
+// typically 500ns of wating on positive/negative edge of CS should be enough ( -> datasheet); to make sure we cover a lot of devices we spend 5ms ( factor 10 !)
+// TODO: analyze if less wating could be sufficient
+#define P039_CS_Delay()             delayMicroseconds(5)
+
+#define P039_MAX_TYPE               PCONFIG(0)
+#define P039_TC_TYPE                PCONFIG(1)
+#define P039_FAM_TYPE               PCONFIG(2)
+#define P039_RTD_TYPE               PCONFIG(3)
+#define P039_RTD_CON_TYPE           PCONFIG(4)
+#define P039_RTD_FILT_TYPE          PCONFIG(5)
+#define P039_RTD_RES                PCONFIG(6)
+#define P039_RTD_LM_TYPE            PCONFIG(7)
+#define P039_RTD_LM_SHTDWN          PCONFIG(8)
```

Comment:
That'll fail (read/write into the next Task's variables), as there are only 8 PCONFIG variables available, meaning they numbered from 0..7.
You can use `PCONFIG_FLOAT(<0..3>)` or `PCONFIG_LONG(<0..3>)` instead (these each have 4 variables available).

Reply:
Thank You for this feedback. 

I guess it was not visible as i did not activate the shutdown option yet. Helpful information !

Reply:
Hmm, maybe we should add some static asserts in the code of that macro that will break the compile if used with values over the defined length.
I wonder if the current code will already break then :)

Reply:
Introduced PCONFIG_LONG[0] fpr P039_RTD_RES to solve the issue - part of latest commit.